File: web/src/main/java/by/sorface/passport/web/controllers/RenewPasswordController.java
package by.sorface.passport.web.controllers;

import by.sorface.passport.web.facade.accounts.RenewPasswordFacade;
import by.sorface.passport.web.records.accounts.AccountRenewPasswordRequest;
import by.sorface.passport.web.records.tokens.ApplyNewPasswordRequest;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/accounts/passwords")
public class RenewPasswordController {

    private final RenewPasswordFacade renewPasswordFacade;

    @PostMapping("/forget")
    public ResponseEntity<?> forgetPassword(@RequestBody @Valid AccountRenewPasswordRequest request) {
        renewPasswordFacade.forgetPassword(request.email());

        return ResponseEntity.ok().build();
    }

    @PostMapping("/new/apply")
    public ResponseEntity<?> applyNewPassword(@RequestBody @Valid ApplyNewPasswordRequest request) {
        renewPasswordFacade.applyNewPassword(request);

        return ResponseEntity.ok().build();
    }

    @GetMapping("/hash/introspect")
    public ResponseEntity<?> checkRenewPasswordToken(@RequestParam("hash") String hash) {
        renewPasswordFacade.checkRenewPasswordToken(hash);

        return ResponseEntity.ok().build();
    }

}

File: web/src/main/kotlin/by/sorface/passport/web/controllers/RenewPasswordController.kt
package by.sorface.passport.web.controllers

import by.sorface.passport.web.facade.accounts.RenewPasswordFacade
import by.sorface.passport.web.records.accounts.AccountRenewPasswordRequest
import by.sorface.passport.web.records.tokens.ApplyNewPasswordRequest
import jakarta.validation.Valid
import lombok.RequiredArgsConstructor
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/accounts/passwords")
class RenewPasswordController {
    private val renewPasswordFacade: RenewPasswordFacade? = null

    @PostMapping("/forget")
    fun forgetPassword(@RequestBody request: @Valid AccountRenewPasswordRequest?): ResponseEntity<*> {
        renewPasswordFacade!!.forgetPassword(request!!.email)

        return ResponseEntity.ok().build<Any>()
    }

    @PostMapping("/new/apply")
    fun applyNewPassword(@RequestBody request: @Valid ApplyNewPasswordRequest?): ResponseEntity<*> {
        renewPasswordFacade!!.applyNewPassword(request!!)

        return ResponseEntity.ok().build<Any>()
    }

    @GetMapping("/hash/introspect")
    fun checkRenewPasswordToken(@RequestParam("hash") hash: String?): ResponseEntity<*> {
        renewPasswordFacade!!.checkRenewPasswordToken(hash)

        return ResponseEntity.ok().build<Any>()
    }
}
